packages/core-plugin-api/src/icons/types.ts

```diff
@@ -31,3 +32,10 @@ import { ComponentType } from 'react';
 export type IconComponent = ComponentType<{
   fontSize?: 'default' | 'small' | 'large';
 }>;
+
+/**
+ * This exists for backwards compatibility with the old core package.
+ * It's used in some parts of this package in order to smooth out the
+ * migration, but it is not exported.
```

Comment:
It is tough! As per the api report above

Reply:
Oh yeah totally, fixed x)